bindings/matrix-sdk-ffi/src/timeline/configuration.rs

```diff
@@ -52,28 +110,27 @@ impl From<DateDividerMode> for matrix_sdk_ui::timeline::DateDividerMode {
 }
 
 #[derive(uniffi::Enum)]
-pub enum AllowedMessageTypes {
+pub enum TimelineFilter {
+    /// Show all the events in the timeline, independent of their type.
     All,
-    Only { types: Vec<RoomMessageEventMessageType> },
+    /// Show only `m.room.messages` of the given room message types.
+    OnlyMessages {
```

Comment:
Nit: I would call this one `OnlyMessage` as `.onlyMessage(types:)` rolls better of the tongue.